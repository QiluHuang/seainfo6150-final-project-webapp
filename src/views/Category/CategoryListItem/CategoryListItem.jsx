import React, { useState } from "react";
import { Switch, Route, Link, useRouteMatch } from "react-router-dom";

import PropTypes from "prop-types";
import styles from "./CategoryListItem.module.css";
// import ArticleTextToggleButton from "../ArticleTextToggleButton/ArticleTextToggleButton.jsx";
import DishImageMedium from "../../../components/Images/DishImageMedium/DishImageMedium.jsx";
import AddButton from '../../../components/Buttons/Button/Button.jsx';
import PlusMinusButton from '../../../components/Buttons/PlusMinusButton/PlusMinusButton.jsx';
import Cart from "../../Cart/Cart";

const DishListItem = (props) => {

    const [cart, setCart] = useState([]);
    const [page, setPage] = useState('dishes');


    const addToCart = (dish) => {
        console.log('we are in addToCart');
        setCart([...cart, dish]);
    }

    console.log(props);
    return (
        <div className={styles.container}>
            <div className={styles.wrapper}>
                <div>
                        <Link to={`/menu/${props.dish.categoryID}/${props.dish.slug}`}>
                            <div className = {styles.upper_image}>
                                <DishImageMedium url={props.dish.image._url} title={props.dish.name}/>
                            </div>
                            <div><h4 className={styles.name}>{props.dish.name}</h4></div>
                        </Link> 
                    </div> 
                    
                        <div><h5 className={styles.text}>{props.dish.ingredients}</h5></div>
                        <div><h4 className={styles.price}>{props.dish.price}</h4></div>
                        <div className={styles.button}><AddButton buttonText="Add" onClick={() => addToCart(props.dish)}/></div>
                    
            </div>
            {/* <Route 
                        path="/cart" 
                        exact
                        render={() => <Cart dishes={cart}/>}/> */}

        </div>
    );
};

DishListItem.propTypes = {
  dishes: PropTypes.object.isRequired,
  onClick: PropTypes.func.isRequired
};
export default DishListItem;