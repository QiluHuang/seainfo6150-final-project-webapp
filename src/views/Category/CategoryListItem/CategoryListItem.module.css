.container {
    width: auto;
    height: 100%;
  }
  
.container a {
  text-decoration: none;
}

.container:hover {
  box-shadow: 0 5px 10px 5px rgba(0, 0, 0, 0.2);
  border-radius: 15px;
}

  
.wrapper {
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  padding-left: 0px;
  text-align: center;
  font-family: Cambria, Cochin, Georgia, Times, 'Times New Roman', serif;

}
 
  
.upper_image {
    text-align: center;
    padding: 10px;
    /* padding-top: 10px;
    padding-bottom: 10px; */
}


.name {
    font-size: 1.6em;
    text-align: center;
    color: black;
}


  
.text {
    font-size: 1.1em;
    margin-left: 10px;
    margin-right: 10px;
}

.price {
  font-size: 1.2em;
}


.button_row {
  display: flex;
  justify-content: center;
  margin-bottom: 20px;
}

.button {
  margin: 10px;
}


@media (max-width: 768px) {
  .name {
    font-size: 1.3em;
  }

  .text {
    font-size: 1em;
  }
}

@media (min-width: 768px) and (max-width: 1390px) {
  .name {
    font-size: 1.4em;
  }

  .text {
    font-size: 1em;
  }
}

