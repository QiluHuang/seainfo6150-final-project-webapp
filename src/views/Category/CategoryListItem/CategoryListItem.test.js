import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import CategoryListItem from "./CategoryListItem.jsx";

const allDishes = {
    "0101-chicken-with-sweet-cream-corn-soup": {
      "slug": "0101-chicken-with-sweet-cream-corn-soup",
      "name": "Chicken With Sweet Cream Corn Soup",
      "categoryID" : "soup",
      "category": "Soup",
      "text": "It’s full of protein and veggies and it’s naturally gluten-free, and it is sooo comforting and delicious.ter.",
      "ingredients": "Chicken, Corn, Ground ginger and garlic powder, Green Onions, Carrots, Cornstarch, Eggs, Toasted sesame oil, Salt and pepper",
      "image": {
        "url": "https://imageresizer.static9.net.au/4X0xzcYzOVZeMKP7lsvoNmdEHz4=/1274x717/smart/https%3A%2F%2Fprod.static9.net.au%2F_%2Fmedia%2FImages%2FKitchen%2F2011%2F09%2F22%2F11%2F34%2Fchickensoup.jpg",
        "_height": "282",
        "_width": "501"
      },
      "price": "$9.99"
    },
    
    "0102-wonton-soup": {
      "slug": "0102-wonton-soup",
      "name": "Wonton Soup",
      "categoryID" : "soup",
      "category": "Soup",
      "text": "Traditionally, wontons are very delicate little things, sometimes filled with vegetables and usually pork…served in a flavorful broth seasoned with some sesame oil and white pepper.",
      "ingredients": "Pork, Shrimp, Green Onion, Ginger",
      "image": {
        "url": "https://www.skinnytaste.com/wp-content/uploads/2020/01/Wonton-Soup-7.jpg",
        "_height": "852",
        "_width": "550"
      },
      "price": "$9.99"
    },
    
    "0103-Egg-Flower-Soup": {
      "slug": "0103-Egg-Flower-Soup",
      "name": "Egg Flower Soup",
      "categoryID" : "soup",
      "category": "Soup",
      "text": "The direct translation in Chinese is egg flower soup, because the egg creates large and small swirls in the soup in a flower-like pattern",
      "ingredients": "Egg, White Pepper, Chicken Stock",
      "image": {
        "url": "https://thewoksoflife.com/wp-content/uploads/2019/05/egg-drop-soup-4.jpg",
        "_height": "1260",
        "_width": "900"
      },
      "price": "$9.99"
    }
}

describe("CategoryListItem tests", () => {
    it("renders the CategoryListItem component correctly", () => {
    const { container } = render(<Router><CategoryListItem dish={allDishes} /></Router>);
    expect(container).toMatchSnapshot();
    });
});