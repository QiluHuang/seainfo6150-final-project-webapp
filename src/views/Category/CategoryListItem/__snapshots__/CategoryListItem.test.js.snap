// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryListItem tests renders the CategoryListItem component correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="wrapper"
    >
      <div>
        <a
          href="/menu/undefined/undefined"
        >
          <div
            class="upper_image"
          />
          <div>
            <h4
              class="name"
            />
          </div>
        </a>
      </div>
      <div>
        <h5
          class="text"
        />
      </div>
      <div>
        <h4
          class="price"
        />
      </div>
      <div
        class="button"
      >
        <button
          class="button"
        >
          Add
        </button>
      </div>
    </div>
  </div>
</div>
`;
